src/utils/utils_assert.h

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ * Copyright (C) 2023-2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#ifndef UMF_ASSERT_H
+#define UMF_ASSERT_H 1
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define NOFUNCTION                                                             \
+    do {                                                                       \
+    } while (0)
+#define VALGRIND_ANNOTATE_NEW_MEMORY(p, s) NOFUNCTION
+#define VALGRIND_HG_DRD_DISABLE_CHECKING(p, s) NOFUNCTION
+
+#ifdef NDEBUG
+#define ASSERT(x) NOFUNCTION
+#define ASSERTne(x, y) ASSERT(x != y)
+#else
+#define ASSERT(x)                                                              \
+    do {                                                                       \
+        if (!(x)) {                                                            \
+            fprintf(stderr,                                                    \
+                    "Assertion failed: " #x " at " __FILE__ " line %d.\n",     \
+                    __LINE__);                                                 \
+            abort();                                                           \
+        }                                                                      \
+    } while (0)
+#define ASSERTne(x, y)                                                         \
+    do {                                                                       \
+        long X = (x);                                                          \
+        long Y = (y);                                                          \
+        if (X == Y) {                                                          \
+            fprintf(stderr,                                                    \
+                    "Assertion failed: " #x " != " #y                          \
+                    ", both are %ld, at " __FILE__ " line %d.\n",              \
+                    X, __LINE__);                                              \
+            abort();                                                           \
+        }                                                                      \
+    } while (0)
+#endif
+
+#define UMF_CHECK(condition, errorStatus)                                      \
+    do {                                                                       \
+        if (!(condition)) {                                                    \
+            fprintf(stderr, "UMF check failed: " #condition " in %s\n",        \
```

Comment:
The question not related to this PR.
Are we going to replace `fprintf` with logger API once we have it? Printing to `stderr` is a temporary solution, right?

Reply:
I think so